### PR TITLE
[4.4] Move to ESM

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes

- Move tools and source JS to ESM
- ES6 files have now the extension `.mjs` (used to be `.es6.js`)
- Web Components also have `.mjs` extension (used to be `.w-c.es6.js`)
- The `build/media_source/system/js/joomla-core-loader.mjs` file now includes the required css and the related `build/media_source/system/scss/joomla-core-loader.scss` is deleted

### Testing Instructions

You need Git and NPM to test this

Pull the `4.4-dev` branch and run:
- `npm install`
- `npm run gzip`
- `npm run versioning`
- Store the media folder somewhere outside of the `joomla-cms` folder

Pull this PR `gh pr checkout 43779` and run:
- `npm install`
- `npm run gzip`
- `npm run versioning`
- Store the media folder somewhere outside of the `joomla-cms` folder

Compare the two folders. The only difference should be the missing files: `media/system/css/joomla-core-loader.css`, `media/system/css/joomla-core-loader.min.css`, `media/system/css/joomla-core-loader.min.css.gz`

Check that basic backend functionality

### Link to documentations
Please select:
- [x] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [x] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

@Fedik @richard67 could you please test this one (pretty hard for the average tester)?

@LadySolveig @bembelimen please DO NOT upmerge this, I will do a PR on the 5.1 instead

### CODE REVIEW

File: build/README.md [hunk 1]
```diff
@@ -5,6 +5,8 @@ Joomla provides a set of tools for managing static assets and dependencies based
 ## Node Based Tools
 The responsibilities of these tools are:
 - To copy files from the `node-modules` folder to the `media` folder.
+- The configuration for the localization of any node packages is held inside the `package.json` on the root folder uder the key `settings`.
```
Comment (brianteeman) [inline build/README.md]:
```suggestion
- The configuration for the localization of any node packages is held inside the `package.json` on the root folder under the key `settings`.
```

typo

**Comment by Fedik** [comment]
Thanks for the work.

I do not like renaming under `/media_source`. We do not have actual modules there, only regular scripts. I would call `.mjs` only for scripts that actualy use ESM (have `export` or `import` in it) There only few of then (even web components not really utilise it). Personaly I would not bother with it.
But I don't mind, if release managers decide to accept it.

For the build script it looks much better there. Because there is actual ESM now.

**Comment by dgrammatiko** [comment]
> For the build script it looks much better there. Because there is actual ESM now.

@Fedik it's just a convention, but now we have way less options, which, normally, should be easier for every contributor

**Comment by Fedik** [comment]
> should be easier for every contributor

yeah

**Comment by Fedik** [comment]
I have tested this item :white_check_mark: successfully on 6b3316ba576993d020d93ba24486be0c4fc09ae6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43779">issues.joomla.org/tracker/joomla-cms/43779</a>.</sub>

**Comment by laoneo** [comment]
Thanks for your hard work on this. Unfortunately 4.4 is in maintenance mode, where we merge mainly bug fixes. To not risk to have the 4.4-dev unstable and prevent side effects in the ecosystem, I'm closing this one for now as the move to ESM should be done in an active development branch, which is currently 5.2. Thanks for understanding.

**Comment by dgrammatiko** [comment]
@laoneo the only reason I backported the changes was to make it easier for devs when they'll have to push a change from 4.4 to 5/6. Anyways np

**Pull request closed without merging**